### PR TITLE
Concurrent Segment Search Blog

### PR DESCRIPTION
**Pull request by sohami**
### Description
This blog talks about the technical details of search path and concurrent search in particular. It also shares some of preliminary performance numbers from `nyc_taxis` workload.
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/2265

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2023-09-25-concurrent_segment_search.markdown [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## Performance results
+
+Currently, the feature is launched as experimental and we don't recommended using it in production. We are working on resolving all the issues found along the way to ensure the feature is stable for production environments. You can track in the status of the feature in the [project board](https://github.com/orgs/opensearch-project/projects/117/views/1) concurrent search section. With the current set of functionality in place, we collected very early benchmark numbers using the OpenSearch Benchmark [`nyc_taxis` workload](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/nyc_taxis). The following sections summarize these benchmarking results.
+
+### Cluster setup 1
+
+* Instance Type: *r5.2xlarge*
```
Comment (natebower) [inline _posts/2023-09-25-concurrent_segment_search.markdown:79]:
```suggestion
* Instance type: *r5.2xlarge*
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and tag me once addressed so that I can verify lines 45, 71, and 75 before approving. Thanks!

File: _posts/2023-09-25-concurrent_segment_search.markdown [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+<img src="/assets/media/blog-images/2023-09-25-concurrent-segment-search/os-search-flow.png" alt="Search request scatter-gather protocol"/>{: .img-fluid}
+
+### Query phase workflow
+
+When a shard receives a search request, it tries to rewrite the query (as needed), creating internal objects such as a query, weights, and a collector tree needed to process this request using Lucene. These represent the various clauses or operations like filters, `terminate_after`, or aggregations defined in the body of the search request. Once the setup is complete, Lucene’s search API is called to perform the search on all the segments of that shard sequentially. 
+
+For each segment, Lucene applies the filter or Boolean clauses to identify and score matching documents. Then the matched documents are passed through the collector tree to collect the matched document IDs. For example, `TopDocsCollector` collects top matching documents, while different `BucketCollector` objects aggregate matching document values depending on the aggregation type.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _posts/2023-09-25-concurrent_segment_search.markdown]: Yes, it is. So there can be either filter clause(s) or Boolean clause(s) in a query.

**Comment by kolchfa-aws** [comment]
@natebower Thanks for the review! I addressed your feedback.

**Review by natebower — APPROVED** [review]
LGTM. @pajuric This is ready to be published. Thanks!

**Comment by Pallavi-AWS** [comment]
Can we get the blog published this week please?

File: _posts/2023-09-25-concurrent_segment_search.markdown [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+title:  "Introducing concurrent segment search in OpenSearch"
+authors: 
+  - reta
+  - sohami
+date:   2023-09-25
+categories: 
+  - search
+  - technical-post
+meta_keywords: concurrent segment search, search performance, search concurrency, OpenSearch 2.10
+meta_description: In OpenSearch 2.10, you can now enable concurrent segment search as an experimental feature in your OpenSearch cluster. With concurrent segment search, you can search through shard segments in parallel and improve search performance for your workload.
```
Comment (pajuric) [inline _posts/2023-09-25-concurrent_segment_search.markdown:12]: Please update the meta with the following: 

**meta_keywords:** Concurrent segment search, search concurrency, concurrent shard search, OpenSearch 2.10
**meta description:** Learn how the experimental concurrent segment search feature in OpenSearch helps you search shard segments in parallel and improves search performance for your workloads.

File: _posts/2023-09-25-concurrent_segment_search.markdown [hunk 4]
```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title:  "Introducing concurrent segment search in OpenSearch"
+authors: 
+  - reta
+  - sohami
+date:   2023-09-25
```
Comment (pajuric) [inline _posts/2023-09-25-concurrent_segment_search.markdown:7]: Please update the date to: 2023-10-18

Comment (pajuric) [reply _posts/2023-09-25-concurrent_segment_search.markdown:7]: @sohami - If you can make these small updates now, I can have the team push your blog live.

**Comment by pajuric** [comment]
@krisfreedain @nateynateynate @dtaivpp - Blog is ready to go. Not sure if you can push the date change and updated meta, but that's the only thing stopping this one from going live.

File: _posts/2023-09-25-concurrent_segment_search.markdown (hunk 4 above)
Comment (krisfreedain) [reply _posts/2023-09-25-concurrent_segment_search.markdown]: done

File: _posts/2023-09-25-concurrent_segment_search.markdown (hunk 3 above)
Comment (krisfreedain) [reply _posts/2023-09-25-concurrent_segment_search.markdown]: done

**Review by krisfreedain — APPROVED** [review]
LGTM!

**Pull request merged**